relayer-cli/src/commands/query/client.rs

```diff
@@ -19,19 +19,25 @@ use crate::conclude::Output;
 use crate::error::{Error, Kind};
 use crate::prelude::*;
 
+// TODO: refactor commands: why is chain_id an `Option`? simpler to give it `ChainId` type.
+//  see the tx/client.rs or tx/packet.rs for simpler approaches.
```

Comment:
That's because `gumdrop` requires types used in options to have a `Default` instance: https://gist.github.com/romac/23db89d7c5971b0e591a11286fc259a6

While this makes sense for options which are not marked `required`, it's beyond me why the requirement subsists for `required` options.

As such, I am not even sure we need the `required` annotation since we end up checking for the `Option` to be defined anyway but I guess it does not hurt if the error message thrown by `gumdrop` is good enough.

Reply:
Oh but we have a `Default` instance for `ChainId`. Then yeah we don't need the option and can instead just use `required`.

Reply:
We added `Default` -s for everything when we started to use types in the CLI options. I think we should consider moving to something else in V1 (maybe directly `gumdrop`) as we need more flexibility.

Reply:
> As such, I am not even sure we need the required annotation since we end up checking for the Option to be defined anyway but I guess it does not hurt if the error message thrown by gumdrop is good enough.

The error message from gumdrop is a bit.. underwhelming. But it helps still. It says only:
> error: missing required free argument

Was looking at ways to parametrize this, but did not find anything.

The way I understand it, with 'required' we now have two levels of checks:

1. gumdrop enforces some (any) value to be present
2. in our validate_options methods we parse and do additional checks on the field.


> We added Default -s for everything when we started to use types in the CLI options. I think we should consider moving to something else in V1 (maybe directly gumdrop) as we need more flexibility.

Sounds like a plan.

Reply:
Opened an issue to continue discussion, if necessary, and track this

https://github.com/informalsystems/ibc-rs/issues/573